support/sier-codec/src/definition_parser.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{
+    schema::{FieldDef, StructDef, Type},
+    Error,
+};
+
+use nom::{
+    bytes::complete::tag,
+    character::complete::{alphanumeric1, multispace0, multispace1},
+    multi::{many0, separated_list0},
+    IResult,
+};
+
+struct ParsedStruct<'i> {
+    type_name: &'i str,
+    fields: Vec<ParsedField<'i>>,
+}
+
+#[derive(Debug)]
+struct ParsedField<'i> {
+    name: &'i str,
+    type_: TypeDef<'i>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum TypeDef<'i> {
+    Primitive(Type),
+    Unresolved(&'i str),
+}
+
+impl<'i> ParsedStruct<'i> {
+    fn compile(self) -> Result<StructDef, Error<'i>> {
+        use std::collections::HashSet;
+
+        let mut seen = HashSet::with_capacity(self.fields.len());
+        for field in &self.fields {
+            dbg!(&field);
```

Comment:
what are the guidelines concerning dbg! statements in to be merged commits, remove them?

Reply:
They should be removed. Done.